### DISCUSSION TITLE
Have  you though of adding an importer for STL file formats

### DISCUSSION THREAD

**Discussion by larrywyre** [FXyz/FXyz#170] [Ideas]
I wanted to be about to support STL file format loading.
The reason is that I dusted off the old (2011) Toxiclibs and have updated it (for my purposes) to support a nice physics model for JavaFX.
I am working on a 3D star mapping system. It's just about ready for Beta and some of the things I am interested in doing is supporting a generalized physics model to use in it for Galaxy evolution and asteroid fields.
I integrated OreKit (https://www.orekit.org) which is an orbital dynamics package but a more generalized physics model that support Verlet Integration would be cool.
I actually thing that using the Physics3d engine in your project would be very cool. They support using a mesh (STL driven) to build all sorts of interesting capabilities.

**Comment by Birdasaur** [comment]
Hey @larrywyre . Sorry about the near year long delayed response. How is this project going for you? Is it still active?
`I am working on a 3D star mapping system. It's just about ready for Beta and some of the things I am interested in doing is supporting a generalized physics model to use in it for Galaxy evolution and asteroid fields.`
This is a very aggressive goal. ;-)

`I integrated OreKit (https://www.orekit.org/) which is an orbital dynamics package`
I'm familiar with using OreKit in Java actually. ;-)

Admittedly I would like to have an STL loader as well. Model loading can be non-trivial due to the details but I'd be interested in helping, especially if we can find a working example for another framework.
Question: What are you using STL models for? The irregular asteroid shapes?

**Comment by Birdasaur** [comment]
One of the reasons we haven't implemented our own STL importer is that its already done by others. See:
https://stackoverflow.com/questions/27581119/import-stl-file-in-javafx
and
https://stackoverflow.com/questions/19462571/how-to-create-3d-shape-from-stl-in-javafx-8
and there are others. I doubt the support for the STL spec is super great but probably good enough.

Also given that we are starting to use issues to branch, I'm converting this to a discussion. It is unlikely that we will be implementing our own STL importer, though we would accept PRs from others whom contribute.
